bravado/client.py

```diff
@@ -120,15 +120,20 @@ def from_spec(cls, spec_dict, origin_url=None, http_client=None,
         :rtype: :class:`bravado_core.spec.Spec`
         """
         http_client = http_client or RequestsClient()
+        config = config or {}
 
         # Apply bravado config defaults
-        config = dict(CONFIG_DEFAULTS, **(config or {}))
+        bravado_config = BravadoConfig.from_config_dict(config)
+        # remove bravado configs from config dict
+        for key in set(bravado_config).intersection(set(config)):
```

Comment:
`bravado_config` is a namedtuple object, calling set on it will return the content of the different fields
```python
>>> C = namedtuple('C', 'a, b, c')
>>> instance = C(1,2,3)
>>> print instance
C(a=1, b=2, c=3)
>>> set(instance)
set([1, 2, 3])
>>> set(C._fields)
set(['a', 'c', 'b'])
>>> 
```